src/main/java/cinema/project/security/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package cinema.project.security;
+
+import cinema.project.model.User;
+import cinema.project.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserDetailsServiceImpl implements UserDetailsService {
+    private final UserService userService;
+
+    public UserDetailsServiceImpl(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userService.findByEmail(email).get();
+        UserBuilder userBuilder = org.springframework.security.core.userdetails.User
```

Comment:
When you create UserBuilder, move package name from method to imports. 
This is the one of the points in the checklist.